code/modules/mob/living/simple_animal/friendly/merchants.dm

```diff
@@ -0,0 +1,45 @@
+/mob/living/simple_animal/merchant
+	name = "Merchant?"
+	desc = "A rock. It seems to have... wares."
+	icon_state = "basalt"
+	icon = 'icons/obj/flora/rocks.dmi'
+	maxHealth = 200
+	health = 200
+	del_on_death = TRUE
+	stop_automated_movement = TRUE
+	loot = list(/obj/effect/decal/cleanable/dirt)
+	///What products are being assigned to the product list?
+	var/list/product_list = list(/obj/item/food/burger/plain = 1, /obj/item/gun/ballistic/automatic/pistol = 1, /obj/item/seeds/rainbow_bunch = 1)
+	///What is the base price for the products being offered?
+	var/product_cost = 100
```

Comment:
I'm not gonna lie, the payment component does not seem like the perfect use-case of this, I can see a lot of situations where people wouldnt want every shop item to have the same price.